[2.9.0] 1373 support authentication with service account tokens (#1402)
rancher/rancher-docs: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md

---
title: JSON Web Token (JWT) Authentication
---

> <!-- Edit Canonical Link -->

Reviewer: ```suggestion

```

---

> title: JWT Authenticaiton

Reviewer 2: Probably should expand the acronym in the title. In-text references expand the acronym on first use, then use JWT throughout

---
<!-- Edit Canonical Link -->
<head> 
  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication"/>
</head>

Many 3rd party integrations available for Kubernetes, such as Gitlab and Hashicorp Vault, involve giving an external process access to the Kubernetes API using a native Kubernetes Service Account token for authentication.

> In Rancher v2.9.0 and later, you can enable a downstream cluster to support JWT authentication of tokens created for a service account on a downstream cluster through the Rancher authentication proxy. Prior to this feature, Rancher rejects such requests because Rancher only supports Rancher-issued tokens.

Reviewer 2: ```suggestion
In Rancher v2.9.0 and later, you can enable a downstream cluster to support JSON web token (JWT) authentication of tokens created for a service account on a downstream cluster through the Rancher authentication proxy. In Rancher versions earlier than  v2.9.0, only Rancher-issued tokens were supported.
```

I'm having some trouble making "you can enable a downstream cluster to support JSON web token (JWT) authentication of tokens created for a service account on a downstream cluster through the Rancher authentication proxy" shorter and easier to read. I feel like this should probably be split into two sentences, or reorganized so there aren't as many clauses. Something like, "service accounts on downstream clusters can now authenticate through JWT using the Rancher authentication proxy", if I'm understanding the sentence correctly.

> :::note
> JWT authentication is also known as service account token authentication.
> :::

Reviewer 2: ```suggestion
JWT authentication is also known as service account token authentication.
```

Is that generally true, or is it just that Rancher uses JWT for service account authentication?

Reviewer 3: Not strictly true.
Service Account Token Auth happens to be JWT-based (and in the original github issue, we call this feature Service account token auth), but to say that JWT auth is a.k.a. Service account token auth would be incorrect.  JWTs have many other uses outside of Rancher or service account token auth.

Author: @crobby Thanks for confirming. I removed this.
